### PR TITLE
docs: Add Minikube deployment to k8s docs

### PR DESCRIPTION
**Pull request by nitisht**
## Description
This PR adds docs on how to deploy Minio on Minikube

## Motivation and Context
This will help users trying to get started with Minio on Kubernetes without having to set a real container cluster. 

## How Has This Been Tested?
Run locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: docs/orchestration/minikube/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Kubernetes running locally on your computer.
+
+## 1. Prerequisites
+
+[Minikube](https://github.com/kubernetes/minikube/blob/master/README.md#installation) and [`kubectl`](https://kubernetes.io/docs/user-guide/prereqs/)
+installed on your system.
+
+## 2. Steps
+
+* Download [`minio_distributed.sh`](https://raw.githubusercontent.com/minio/minio/master/docs/orchestration/minikube/minio_distributed.sh) and [`statefulset.yaml`](https://raw.githubusercontent.com/minio/minio/master/docs/orchestration/minikube/statefulset.yaml) files on your computer.
```
Comment (deekoder) [inline docs/orchestration/minikube/README.md]: You could simply give a wget command for easy cut copy paste.

File: docs/orchestration/minikube/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+* Execute the `minio_distributed.sh` script in command prompt. After the script is executed successfully, you should get an output like this
+
```
Comment (deekoder) [inline docs/orchestration/minikube/README.md]: It's better to show than ask user to execute. Makes for easy cut copy paste.

File: docs/orchestration/minikube/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+Minio server can now be accessed at `http:localhost:9000`, with accessKey and secretKey as mentioned in the `statefulset.yaml` file.
+
+## 3. Notes
+
+Minikube currently does not support dynamic provisioning, so we manually create PersistentVolumes(PV) and PersistentVolumeClaims(PVC) to be
+used by the distributed Minio setup. Once the PVs and PVCs are created, we call the `statefulset.yaml` configuration file to create the distributed Minio setup.
+
+This setup is supposed to run on a laptop/computer. Hence only one disk is used as the backend for all the minio instance PVs. Minio sees these PVs as separate disks and reports the available storage incorrectly.
```
Comment (deekoder) [inline docs/orchestration/minikube/README.md]: "supposed" to run sounds like we are speculating. We should sound more definitive IMO :)

**Review by deekoder — CHANGES_REQUESTED** [review]

File: docs/orchestration/minikube/README.md (hunk 1 above)
Comment (nitisht) [reply docs/orchestration/minikube/README.md]: done.

File: docs/orchestration/minikube/README.md (hunk 2 above)
Comment (nitisht) [reply docs/orchestration/minikube/README.md]: done.

File: docs/orchestration/minikube/README.md (hunk 3 above)
Comment (nitisht) [reply docs/orchestration/minikube/README.md]: I agree @deekoder . Removed `supposed`

**Comment by nitisht** [comment]
Thanks for the review @deekoder. Addressed the comments, please take a look.

**Comment by deekoder** [comment]
will wrap this up tomorrow. ETA : Apt 21 EOD

File: docs/orchestration/kubernetes/README.md [hunk 4]
```diff
@@ -1,15 +1,20 @@
 # Deploy Minio on Kubernetes [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io) [![Go Report Card](https://goreportcard.com/badge/minio/minio)](https://goreportcard.com/report/minio/minio) [![Docker Pulls](https://img.shields.io/docker/pulls/minio/minio.svg?maxAge=604800)](https://hub.docker.com/r/minio/minio/) [![codecov](https://codecov.io/gh/minio/minio/branch/master/graph/badge.svg)](https://codecov.io/gh/minio/minio)
 
-Kubernetes constructs like Deployments and StatefulSets provide perfect platform to deploy Minio server in standalone, distributed or shared mode. In addition, using Minio [Helm](https://helm.sh) Chart, you can deploy Minio server with a single command on your cluster. 
+Kubernetes constructs like Deployments and StatefulSets provide perfect platform to deploy Minio server in standalone, distributed or shared mode. There are multiple options to deploy Minio on Kubernetes, you can choose the one that suits you the most.
```
Comment (deekoder) [inline docs/orchestration/kubernetes/README.md]: constructs like.. sounds odd to me. I don't have kubernetes domain expertise. Just wanted to check.

File: docs/orchestration/kubernetes/README.md [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
 
-Minio Helm Chart offers great deal of [customizability](#configuration), still if you'd rather like to deploy Minio using custom config files, you can do that as well. This [blog post](https://blog.minio.io/build-aws-s3-compatible-cloud-storage-on-gcp-with-minio-and-kubernetes-159cc99caea8#.8zesfh6tc) offers an introduction to running Minio on Kubernetes using .yaml configuration files.
+- Minio [Helm](https://helm.sh) Chart offers a customizeable yet easy Minio deployment, with a single command. Read more about Minio Helm deployment [here](#prerequisites).
```
Comment (deekoder) [inline docs/orchestration/kubernetes/README.md]: Minio Helm Chart offers a customizable (american spelling) and easy Minio deployment with a single command.

File: docs/orchestration/kubernetes/README.md [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
 
+- You can also explore Kubernetes [Minio example](https://github.com/kubernetes/kubernetes/blob/master/examples/storage/minio/README.md) to deploy Minio using `.yaml` files.
+
+- If you'd like get started with Minio on Kubernetes without having to create a real container cluster, you can also [deploy Minio locally](https://raw.githubusercontent.com/minio/minio/master/docs/orchestration/minikube/README.md) with MiniKube.
```
Comment (deekoder) [inline docs/orchestration/kubernetes/README.md]: If you'd like *to* get started...

File: docs/orchestration/minikube/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+```
+kubectl port-forward minio-0 9000:9000
+```
+
+Minio server can now be accessed at `http://localhost:9000`, with accessKey and secretKey as mentioned in the `statefulset.yaml` file.
+
+## 3. Notes
+
+Minikube currently does not support dynamic provisioning, so we manually create PersistentVolumes(PV) and PersistentVolumeClaims(PVC) to be
```
Comment (deekoder) [inline docs/orchestration/minikube/README.md]: ..to be used by.. could it be refactored.

**Review by deekoder — CHANGES_REQUESTED** [review]

**Comment by nitisht** [comment]
Addressed the comments @deekoder

**Comment by deekoder** [comment]
I am trying it out now on my laptop.

**Comment by harshavardhana** [comment]
> I am trying it out now on my laptop.

Is this validation complete @deekoder ? any surprises?

**Comment by deekoder** [comment]
it works i believe.

**Review by deekoder — APPROVED** [review]

**Pull request merged**